{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(TCGAbiolinks)\n",
    "library(SummarizedExperiment)\n",
    "library(edgeR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "--------------------------------------\n",
      "\n",
      "o GDCquery: Searching in GDC database\n",
      "\n",
      "--------------------------------------\n",
      "\n",
      "Genome of reference: hg19\n",
      "\n",
      "--------------------------------------------\n",
      "\n",
      "oo Accessing GDC. This might take a while...\n",
      "\n",
      "--------------------------------------------\n",
      "\n",
      "ooo Project: TCGA-BRCA\n",
      "\n",
      "--------------------\n",
      "\n",
      "oo Filtering results\n",
      "\n",
      "--------------------\n",
      "\n",
      "ooo By platform\n",
      "\n",
      "ooo By experimental.strategy\n",
      "\n",
      "ooo By data.type\n",
      "\n",
      "ooo By file.type\n",
      "\n",
      "ooo By barcode\n",
      "\n",
      "----------------\n",
      "\n",
      "oo Checking data\n",
      "\n",
      "----------------\n",
      "\n",
      "ooo Check if there are duplicated cases\n",
      "\n",
      "ooo Check if there results for the query\n",
      "\n",
      "-------------------\n",
      "\n",
      "o Preparing output\n",
      "\n",
      "-------------------\n",
      "\n",
      "--------------------------------------\n",
      "\n",
      "o GDCquery: Searching in GDC database\n",
      "\n",
      "--------------------------------------\n",
      "\n",
      "Genome of reference: hg19\n",
      "\n",
      "--------------------------------------------\n",
      "\n",
      "oo Accessing GDC. This might take a while...\n",
      "\n",
      "--------------------------------------------\n",
      "\n",
      "ooo Project: TCGA-COAD\n",
      "\n",
      "--------------------\n",
      "\n",
      "oo Filtering results\n",
      "\n",
      "--------------------\n",
      "\n",
      "ooo By platform\n",
      "\n",
      "ooo By experimental.strategy\n",
      "\n",
      "ooo By data.type\n",
      "\n",
      "ooo By file.type\n",
      "\n",
      "ooo By barcode\n",
      "\n",
      "----------------\n",
      "\n",
      "oo Checking data\n",
      "\n",
      "----------------\n",
      "\n",
      "ooo Check if there are duplicated cases\n",
      "\n",
      "ooo Check if there results for the query\n",
      "\n",
      "-------------------\n",
      "\n",
      "o Preparing output\n",
      "\n",
      "-------------------\n",
      "\n",
      "Downloading data for project TCGA-BRCA\n",
      "\n",
      "GDCdownload will download 11 files. A total of 16.610993 MB\n",
      "\n",
      "Downloading as: Tue_Apr__6_09_31_14_2021.tar.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading: 6.7 MB       "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Downloading data for project TCGA-COAD\n",
      "\n",
      "GDCdownload will download 11 files. A total of 16.59348 MB\n",
      "\n",
      "Downloading as: Tue_Apr__6_09_31_39_2021.tar.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading: 6.6 MB     "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-------------------\n",
      "\n",
      "oo Reading 11 files\n",
      "\n",
      "-------------------\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "|====================================================|100%                      Completed after 0 s \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-------------------\n",
      "\n",
      "oo Merging 11 files\n",
      "\n",
      "-------------------\n",
      "\n",
      "Accessing grch37.ensembl.org to get gene information\n",
      "\n",
      "Downloading genome information (try:0) Using: Human genes (GRCh37.p13)\n",
      "\n",
      "Cache found\n",
      "\n",
      "Starting to add information to samples\n",
      "\n",
      " => Add clinical information to samples\n",
      "\n",
      " => Adding TCGA molecular information from marker papers\n",
      "\n",
      " => Information will have prefix 'paper_' \n",
      "\n",
      "brca subtype information from:doi.org/10.1016/j.ccell.2018.03.014\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# list of hotspot and wild-type aliquot ids in brca and coad with experession data\n",
    "listsamples_br <- c(\"TCGA-A7-A13E-01A\",\"TCGA-A7-A0CH-01A\",\"TCGA-A2-A0CX-01A\",\"TCGA-C8-A137-01A\",\"TCGA-AR-A0TP-01A\",\"TCGA-AO-A12F-01A\",\"TCGA-A2-A0SW-01A\",\"TCGA-BH-A0DV-01A\",\"TCGA-C8-A12O-01A\",\"TCGA-C8-A12M-01A\",\"TCGA-A2-A0T5-01A\")\n",
    "listsamples_cd <- c(\"TCGA-4N-A93T-01A\",\"TCGA-AA-A01X-01A\",\"TCGA-NH-A6GA-01A\",\"TCGA-A6-A567-01A\",\"TCGA-AD-6889-01A\",\"TCGA-AA-A01P-01A\",\"TCGA-AA-A022-01A\",\"TCGA-NH-A8F8-01A\",\"TCGA-QG-A5YV-01A\",\"TCGA-QL-A97D-01A\",\"TCGA-DM-A28G-01A\",\"TCGA-AY-A71X-01A\")\n",
    "\n",
    "# query the file for primary tumors\n",
    "\n",
    "query <- GDCquery(project = \"TCGA-BRCA\",\n",
    "                  data.category = \"Gene expression\",\n",
    "                  data.type = \"Gene expression quantification\",\n",
    "                  experimental.strategy = \"RNA-Seq\",\n",
    "                  platform = \"Illumina HiSeq\",\n",
    "                  file.type = \"results\",\n",
    "                  barcode = listsamples_br, \n",
    "                  legacy = TRUE)\n",
    "\n",
    "query2 <- GDCquery(project = \"TCGA-COAD\",\n",
    "                  data.category = \"Gene expression\",\n",
    "                  data.type = \"Gene expression quantification\",\n",
    "                  experimental.strategy = \"RNA-Seq\",\n",
    "                  platform = \"Illumina HiSeq\",\n",
    "                  file.type = \"results\",\n",
    "                  barcode = listsamples_cd, \n",
    "                  legacy = TRUE)\n",
    "# Download the data\n",
    "GDCdownload(query)\n",
    "GDCdownload(query2)\n",
    "\n",
    "####BRCA####\n",
    "\n",
    "# Prepare expression matrix with geneID in the rows and samples (barcode) in the columns\n",
    "# rsem.genes.results as values\n",
    "\n",
    "BRCARnaseqSE <- GDCprepare(query)\n",
    "\n",
    "BRCAMatrix <- assay(BRCARnaseqSE,\"raw_count\") # or BRCAMatrix <- assay(BRCARnaseqSE,\"raw_count\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "I Need about  2.7 seconds for this Complete Normalization Upper Quantile  [Processing 80k elements /s]  \n",
      "\n",
      "Step 1 of 4: newSeqExpressionSet ...\n",
      "\n",
      "Step 2 of 4: withinLaneNormalization ...\n",
      "\n",
      "Step 3 of 4: betweenLaneNormalization ...\n",
      "\n",
      "Step 4 of 4: exprs ...\n",
      "\n",
      "Batch correction skipped since no factors provided\n",
      "\n",
      "----------------------- DEA -------------------------------\n",
      "\n",
      "there are Cond1 type Wild in  5 samples\n",
      "\n",
      "there are Cond2 type Hotspot in  6 samples\n",
      "\n",
      "there are  14892 features as miRNA or genes \n",
      "\n",
      "I Need about  5.5 seconds for this DEA. [Processing 30k elements /s]  \n",
      "\n",
      "----------------------- END DEA -------------------------------\n",
      "\n",
      "-------------------\n",
      "\n",
      "oo Reading 11 files\n",
      "\n",
      "-------------------\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "|====================================================|100%                      Completed after 0 s \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-------------------\n",
      "\n",
      "oo Merging 11 files\n",
      "\n",
      "-------------------\n",
      "\n",
      "Accessing grch37.ensembl.org to get gene information\n",
      "\n",
      "Downloading genome information (try:0) Using: Human genes (GRCh37.p13)\n",
      "\n",
      "Cache found\n",
      "\n",
      "Starting to add information to samples\n",
      "\n",
      " => Add clinical information to samples\n",
      "\n",
      " => Adding TCGA molecular information from marker papers\n",
      "\n",
      " => Information will have prefix 'paper_' \n",
      "\n",
      "coad subtype information from:doi:10.1038/nature11252\n",
      "\n",
      "I Need about  2.7 seconds for this Complete Normalization Upper Quantile  [Processing 80k elements /s]  \n",
      "\n",
      "Step 1 of 4: newSeqExpressionSet ...\n",
      "\n",
      "Step 2 of 4: withinLaneNormalization ...\n",
      "\n",
      "Step 3 of 4: betweenLaneNormalization ...\n",
      "\n",
      "Step 4 of 4: exprs ...\n",
      "\n",
      "Batch correction skipped since no factors provided\n",
      "\n",
      "----------------------- DEA -------------------------------\n",
      "\n",
      "there are Cond1 type Wild in  5 samples\n",
      "\n",
      "there are Cond2 type Hotspot in  6 samples\n",
      "\n",
      "there are  14890 features as miRNA or genes \n",
      "\n",
      "I Need about  5.5 seconds for this DEA. [Processing 30k elements /s]  \n",
      "\n",
      "----------------------- END DEA -------------------------------\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# For gene expression if you need to see a boxplot correlation and AAIC plot to define outliers you can run\n",
    "BRCARnaseq_CorOutliers <- TCGAanalyze_Preprocessing(BRCARnaseqSE)\n",
    "\n",
    "\n",
    "dataNorm <- TCGAanalyze_Normalization(tabDF = BRCARnaseq_CorOutliers, geneInfo =  geneInfo)\n",
    "\n",
    "dataFilt <- TCGAanalyze_Filtering(tabDF = dataNorm,\n",
    "                                  method = \"quantile\", \n",
    "                                  qnt.cut =  0.25)\n",
    "\n",
    "ht=c(\"TCGA-A7-A13E-01A-11R-A12P-07\",\"TCGA-A2-A0CX-01A-21R-A00Z-07\",\"TCGA-C8-A137-01A-11R-A115-07\", \"TCGA-AR-A0TP-01A-11R-A084-07\",\"TCGA-A2-A0SW-01A-11R-A084-07\",  \"TCGA-C8-A12O-01A-11R-A115-07\")\n",
    "\n",
    "wt=c(\"TCGA-A7-A0CH-01A-21R-A00Z-07\",\"TCGA-C8-A12M-01A-11R-A115-07\",\"TCGA-AO-A12F-01A-11R-A115-07\",\"TCGA-BH-A0DV-01A-21R-A12P-07\",\"TCGA-A2-A0T5-01A-21R-A084-07\")\n",
    "\n",
    "\n",
    "dataDEGs <- TCGAanalyze_DEA(mat1 = dataFilt[,wt],\n",
    "                            mat2 = dataFilt[,ht],\n",
    "                            Cond1type = \"Wild\",\n",
    "                            Cond2type = \"Hotspot\",\n",
    "                            method = \"glmLRT\")\n",
    "\n",
    "dataDEGsFiltLevel <- TCGAanalyze_LevelTab(dataDEGs,\"Hotspot\",\"Wild\",\n",
    "                                          dataFilt[,ht],dataFilt[,wt])\n",
    "\n",
    "write.table(dataDEGsFiltLevel, file=\"Results/BRCA_DEG_HW_raw.tsv\", sep=\"\\t\")\n",
    "\n",
    "#### COAD####\n",
    "\n",
    "COADRnaseqSE <- GDCprepare(query2)\n",
    "COADMatrix <- assay(COADRnaseqSE,\"raw_count\") # or COADMatrix <- assay(COADRnaseqSE,\"raw_count\")\n",
    "\n",
    "# For gene expression if you need to see a boxplot correlation and AAIC plot to define outliers you can run\n",
    "COADRnaseq_CorOutliers <- TCGAanalyze_Preprocessing(COADRnaseqSE)\n",
    "\n",
    "\n",
    "dataNorm <- TCGAanalyze_Normalization(tabDF = COADRnaseq_CorOutliers, geneInfo =  geneInfo)\n",
    "\n",
    "dataFilt <- TCGAanalyze_Filtering(tabDF = dataNorm,\n",
    "                                  method = \"quantile\", \n",
    "                                  qnt.cut =  0.25)\n",
    "\n",
    "ht=c(\"TCGA-4N-A93T-01A-11R-A37K-07\", \"TCGA-NH-A6GA-01A-11R-A37K-07\",\"TCGA-A6-A567-01A-31R-A28H-07\" , \"TCGA-NH-A8F8-01A-72R-A41B-07\" ,\"TCGA-QG-A5YV-01A-11R-A28H-07\" ,\"TCGA-QL-A97D-01A-12R-A41B-07\")\n",
    "wt=c(\"TCGA-AA-A01X-01A-21R-A083-07\",\"TCGA-AD-6889-01A-11R-1928-07\",\"TCGA-AA-A01P-01A-21R-A083-07\" ,\"TCGA-DM-A28G-01A-11R-A16W-07\" ,\"TCGA-AY-A71X-01A-12R-A37K-07\")\n",
    "\n",
    "dataDEGs <- TCGAanalyze_DEA(mat1 = dataFilt[,wt],\n",
    "                            mat2 = dataFilt[,ht],\n",
    "                            Cond1type = \"Wild\",\n",
    "                            Cond2type = \"Hotspot\",\n",
    "                            method = \"glmLRT\")\n",
    "\n",
    "dataDEGsFiltLevel <- TCGAanalyze_LevelTab(dataDEGs,\"Hotspot\",\"Wild\",\n",
    "                                          dataFilt[,ht],dataFilt[,wt])\n",
    "\n",
    "write.table(dataDEGsFiltLevel, file=\"Results/COAD_DEG_HW_raw.tsv\", sep=\"\\t\")\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
